### ISSUE TITLE
No possibility to not use tooltip for PopupMenuButton

### ISSUE THREAD

**Issue by rbozan** [flutter/flutter#60418]
We seem to be forced to use a tooltip message even when we don't want to for PopupMenuButton.

https://github.com/flutter/flutter/blob/2962912d68fb72d6c9fd60f652e27f3593d692d6/packages/flutter/lib/src/material/popup_menu.dart#L1151

**Comment by TahaTesser** [comment]
Hi @rbozan 
Is there any particular reason you want to remove tooltip?  
Some material widgets come with tooltip by default, it's by design and i don't see any reason to disable them 

![flutter_02](https://user-images.githubusercontent.com/48603081/86002270-bfa86400-ba2d-11ea-9255-e9b932c6f784.png)
![flutter_03](https://user-images.githubusercontent.com/48603081/86002277-c1722780-ba2d-11ea-8d56-213123ed4b2b.png)

Thank you

**Comment by rbozan** [comment]
It's not really necessary for me, however I am not using the 3 dots icon but using an avatar as the child, and having a tooltip 'Show menu' for that avatar is just not fitting in my opinion.

**Comment by ramnavan** [comment]
While I agree that there should be an option to not show tooltip altogether, I also understand the value in having a default tooltip string. The real issue here is that `null` value has more than one meaning. 1) user has not set and hence resort to default. 2) user does not want tooltip. 

What do people think about having `disableToolTip` property in `PopupMenuButton` widget that defaults to `false`. When it's explicitly set to `true`, it will not render tooltip widget and also ignore `tooltip` value or have assertion checks. 

The same pattern can be followed for other material widgets as well where applicable.

**Comment by jhourlad** [comment]
+1 Definitely a necessity.

**Comment by mlf2020** [comment]
yeah，this property is not commonly used。think about "disableToolTip" property

**Comment by casetofon** [comment]
> Hi @rbozan
> Is there any particular reason you want to remove tooltip?
> Some material widgets come with tooltip by default, it's by design and i don't see any reason to disable them
> 
> ![flutter_02](https://user-images.githubusercontent.com/48603081/86002270-bfa86400-ba2d-11ea-9255-e9b932c6f784.png)
> ![flutter_03](https://user-images.githubusercontent.com/48603081/86002277-c1722780-ba2d-11ea-8d56-213123ed4b2b.png)
> 
> Thank you

Yes there is a reason: I don't want the tooltip there. What other reason do I need? It's my app. If you don't see any reason to disable them, that doesn't mean other people can't have their own reasons... The ego on this guy...

**Comment by Baluds** [comment]
one of the crucial  reason is when we use popupmenubutton in flutter web , whenever you hover over it tooltip text appears which is not required so is there any way to disable them ,it actually takes a lot of space while displaying and in flutter web just hovering will activate it so making it triublesome.

**Comment by JiriTrecak** [comment]
Any update on this? This is rather blocking us from finishing our cloud part of the app. I believe only very few companies will want to show the tooltips on the web, it is very non-standard behavior, especially with tooltips taking such a massive amount of space. Customer: Supernova / Nebula

**Comment by rfrankspk** [comment]
I agree this would be really nice to have, it is very convenient to use PopupMenuButton to show a menu and not necessarily want a tooltip.

**Comment by jomiplaz** [comment]
I have temporarily fixed it by adding the following parent to the PopupMenuButton Widget and setting tooltip to "":
```
Theme(
        data: Theme.of(context).copyWith(
          tooltipTheme: TooltipThemeData(
            decoration: BoxDecoration(
              color: Colors.transparent,
            ),
          )
        ),
        child: PopupMenuButton(
          tooltip: "",
          ...
```

**Comment by rydmike** [comment]
I totally agree on this issue. Widgets that implements tooltips should provide a way to enable/disable tooltips. They should also provide means to change the tooltip content.

>Many web and desktop apps may want to implement a global user controllable feature that enables and disables tooltips in the app entirely. Tooltips are nice for beginners when you are not familiar with the app, but when you know the app they tend to get annoying and get in the way of using the app. Providing users a feature where you can turn off tooltips is a nice way to increase the UX for more experienced users of your app.
>Currently many widget in Flutter that uses tooltips make it impossible (or at least more difficult than it needs to be) to implement this feature, as they have no way of turning off tooltips at all. The `BottomNavigationBar` is another critical widget that fails in this respect.

I reported this sister topic concerning similar issue with the `BottomNavigationBar` https://github.com/flutter/flutter/issues/71049 it was previously mentioned as a part of this proposal https://github.com/flutter/flutter/issues/66993, that would also provide a means of enabling/disabling tooltips globally in an app.

**Comment by rbozan** [comment]
> I totally agree on this issue. Widgets that implements tooltips should provide a way to enable/disable tooltips. They should also provide means to change the tooltip content.
> 
> > Many web and desktop apps may want to implement a global user controllable feature that enables and disables tooltips in the app entirely. Tooltips are nice for beginners when you are not familiar with the app, but when you know the app they tend to get annoying and get in the way of using the app. Providing users a feature where you can turn off tooltips is a nice way to increase the UX for more experienced users of your app.
> > Currently many widget in Flutter that uses tooltips make it impossible (or at least more difficult than it needs to be) to implement this feature, as they have no way of turning off tooltips at all. The `BottomNavigationBar` is another critical widget that fails in this respect.
> 
> I reported this sister topic concerning similar issue with the `BottomNavigationBar` #71049 it was previously mentioned as a part of this proposal #66993, that would also provide a means of enabling/disabling tooltips globally in an app.

Good job in explaining this issue in a more detailed manner :)

**Comment by ggdream** [comment]
> I have temporarily fixed it by adding the following parent to the PopupMenuButton Widget and setting tooltip to "":
> 
> ```
> Theme(
>         data: Theme.of(context).copyWith(
>           tooltipTheme: TooltipThemeData(
>             decoration: BoxDecoration(
>               color: Colors.transparent,
>             ),
>           )
>         ),
>         child: PopupMenuButton(
>           tooltip: "",
>           ...
> ```

It's ok, but i'm lazy, so... I've commented out the source code
~~~dart
      // tooltip: widget.tooltip ?? MaterialLocalizations.of(context).showMenuTooltip,
~~~

**Comment by markusaksli-nc** [comment]
There was some discussion on https://github.com/flutter/flutter/issues/66993 about this wider problem of not being able to disable tooltips and I submitted https://github.com/flutter/flutter/pull/91609 as a proposed implementation of the solution (disabling tooltips in subtrees)

**Comment by ggdream** [comment]
Sounds great, I hope to see it in a stable version soon.
